### PR TITLE
(5/n) Data Refactor - Docs

### PR DESCRIPTION
**Pull request by awaelchli**
Part of https://github.com/Lightning-AI/lit-gpt/issues/954

### CODE REVIEW

File: lit_gpt/data/alpaca.py [hunk 1]
```diff
@@ -109,6 +109,16 @@ def val_dataloader(self) -> DataLoader:
             collate_fn=get_sft_collate_fn(max_seq_length=self.max_seq_length, ignore_index=self.ignore_index)
         )
 
+    def __repr__(self) -> str:
```
Comment (carmocca) [inline lit_gpt/data/alpaca.py:112]: maybe not for this PR, but we should support datamodule dataclasses: https://github.com/Lightning-AI/pytorch-lightning/pull/9039

**Review by carmocca — APPROVED** [review]

File: tutorials/finetune_adapter.md [hunk 2]
```diff
@@ -50,20 +45,20 @@ For example, the following settings will let you finetune the model in under 1 h
 This script will save checkpoints periodically to the `out_dir` directory. If you are finetuning different models or on your own dataset, you can specify an output directory with your preferred name:
 
 ```bash
-python finetune/adapter.py --io.out_dir out/adapter/my-model-finetuned
+python finetune/adapter.py --data Alpaca --io.out_dir out/adapter/my-model-finetuned
```
Comment (rasbt) [inline tutorials/finetune_adapter.md:48]:
```suggestion
python finetune/adapter.py \
  --data Alpaca \
  --io.out_dir out/adapter/my-model-finetuned
```

File: tutorials/finetune_adapter.md [hunk 3]
```diff
@@ -50,20 +45,20 @@ For example, the following settings will let you finetune the model in under 1 h
...
 ```
 
 or for Adapter V2
 
 ```bash
-python finetune/adapter_v2.py --io.out_dir out/adapter_v2/my-model-finetuned
+python finetune/adapter_v2.py --data Alpaca --io.out_dir out/adapter_v2/my-model-finetuned
```
Comment (rasbt) [inline tutorials/finetune_adapter.md:54]:
```suggestion
python finetune/adapter_v2.py \
  --data Alpaca \
  --io.out_dir out/adapter_v2/my-model-finetuned
```

File: tutorials/finetune_adapter.md [hunk 4]
```diff
@@ -50,20 +45,20 @@ For example, the following settings will let you finetune the model in under 1 h
...
 ```
 
 If your GPU does not support `bfloat16`, you can pass the `--precision 32-true` argument.
 For instance, to fine-tune on MPS (the GPU on modern Macs), you can run
 
 ```bash
-python finetune/adapter.py --io.out_dir out/adapter/my-model-finetuned --precision 32-true
+python finetune/adapter.py --data Alpaca --io.out_dir out/adapter/my-model-finetuned --precision 32-true
```
[the remark this answers is not in the document]
Comment (rasbt) [reply tutorials/finetune_adapter.md:61]: Not related to this PR, but actually, I think 16-mixed might work on Macs too?

File: tutorials/finetune_full.md [hunk 5]
```diff
@@ -8,38 +8,33 @@ The steps here only need to be done once:
...
 
+Lit-GPT provides common datasets for finetuning, such as Alpaca, LIMA, Dolly, and more.
+You can optionally [prepare your own dataset](#tune-on-your-dataset).
 For more information about dataset preparation, also see the [prepare_dataset.md](./prepare_dataset.md) tutorial.
 
 ## Running the finetuning
 
 ```bash
-python finetune/full.py --io.checkpoint_dir checkpoints/tiiuae/falcon-7b
+python finetune/full.py --data Alpaca --io.checkpoint_dir checkpoints/tiiuae/falcon-7b
```
Comment (rasbt) [inline tutorials/finetune_full.md:19]:
```suggestion
python finetune/full.py \
  --data Alpaca \
  --io.checkpoint_dir checkpoints/tiiuae/falcon-7b
```

File: tutorials/finetune_full.md [hunk 6]
```diff
@@ -8,38 +8,33 @@ The steps here only need to be done once:
...
 
-You can speed up training by setting the `devices` variable in the script to utilize more GPUs if available.
+You can speed up training by passing the `devices` argument to the script to utilize more GPUs if available.
 Depending on the available GPU memory, you can also tune the `micro_batch_size` parameter to utilize the GPU efficiently.
 
 This script will save checkpoints periodically to the `out_dir` directory. If you are finetuning different models or on your own dataset, you can specify an output directory with your preferred name:
 
 ```bash
-python finetune/full.py --io.out_dir out/full/my-model-finetuned
+python finetune/full.py --data Alpaca --io.out_dir out/full/my-model-finetuned
```
Comment (rasbt) [inline tutorials/finetune_full.md:30]:
```suggestion
python finetune/full.py \
  --data Alpaca \
  --io.out_dir out/full/my-model-finetuned
```

**Comment by awaelchli** [comment]
Thank you for the suggestions 🎉

**Pull request merged**